sycl/include/sycl/ext/oneapi/experimental/reduction_properties.hpp

```diff
@@ -83,9 +97,17 @@ template <typename BinaryOperation>
 struct IsDeterministicOperator<DeterministicOperatorWrapper<BinaryOperation>>
     : std::true_type {};
 
+template <typename PropertyList>
+inline constexpr bool is_valid_reduction_prop_list =
+    ext::oneapi::experimental::detail::all_are_properties_of_v<
+        ext::oneapi::experimental::detail::reduction_property_check_anchor,
+        PropertyList>;
+
 } // namespace detail
 
-template <typename BufferT, typename BinaryOperation, typename PropertyList>
+template <typename BufferT, typename BinaryOperation, typename PropertyList,
+          typename = std::enable_if_t<
+              detail::is_valid_reduction_prop_list<PropertyList>>>
 auto reduction(BufferT vars, handler &cgh, BinaryOperation combiner,
                PropertyList properties) {
```

Comment:
I'm tempted to say I would prefer a `static_assert` on whether the properties are valid. That way we can control the message telling the user why they're not allowed to make the call they're trying to do.

Reply:
I don't that's valid in this case.  The specification says that this is a _Constraint_, and [ISO C++ says](https://eel.is/c++draft/structure.specifications#3.1) (_emphasis mine_):
> [Note [1](https://eel.is/c++draft/structure.specifications#note-1): Failure to meet such a condition results in the function's _silent_ non-viability[.](https://eel.is/c++draft/structure.specifications#3.1.sentence-2) — end note]

We could do this if we changed from a _Constraint_ to a _Mandate_, but I don't know enough about why to favor one over the other.